react-paw-mailmerge/src/components/ViewOutput.tsx

```diff
@@ -1,13 +1,21 @@
 import React from "react";
-import { CsvResult } from "../Helpers/CsvFunctions";
+import { CsvResult, DoMailMerge } from "../Helpers/CsvFunctions";
 import TableOutput from "./table/TableOutput";
+import { ZipAndDownload } from "../Helpers/ZipFunctions";
+import Button from "react-bootstrap/Button"
 
 interface ViewOutputProps {
     parsedData: CsvResult | null;
     template: string;
 }
 
 const ViewOutput: React.FC<ViewOutputProps> = ({parsedData, template}) => {
+    const handleDownloadAll = () => {
+        if (parsedData !== null) {
+            ZipAndDownload(parsedData.data.map((row) => DoMailMerge(row, template)));
```

Comment:
Going to make a ticket just in case we need to track this, but we're going to need to react to the error string also being returned.